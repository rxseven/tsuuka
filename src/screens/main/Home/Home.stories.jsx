import React from 'react';
import { storiesOf } from '@storybook/react';

import Main from 'components/core/sections/Main';
import Wrapper from 'stories/decorators/Wrapper';
import Home from './index';

const title = 'Screens/Main/Home';
const component = <Home />;

storiesOf(title, module)
  .addDecorator(story => (
    <Wrapper>
      <Main>{story()}</Main>
    </Wrapper>
  ))
  .add('default', () => component);
