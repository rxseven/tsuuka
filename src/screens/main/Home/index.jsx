import React from 'react';

import { Body, Document, Head } from 'components/common/base/Page';
import Layout from 'components/common/composite/Layout';
import Currency from 'components/common/modules/Currency';

function Home() {
  return (
    <Document>
      <Head>
        <title>通貨換算ツール</title>
        <meta
          content="An extremely simple React app for converting currencies"
          name="description"
        />
      </Head>
      <Body>
        <Layout>
          <Currency />
        </Layout>
      </Body>
    </Document>
  );
}

export default Home;
