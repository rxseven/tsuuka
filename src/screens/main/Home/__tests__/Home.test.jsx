import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Environment from 'tests/environment';
import Home from '../index';

describe('<Home />', () => {
  // Arrange
  const component = (
    <Environment>
      <Home />
    </Environment>
  );

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
