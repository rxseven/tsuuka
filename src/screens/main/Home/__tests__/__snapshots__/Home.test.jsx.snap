// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Snapshot tests should render correctly 1`] = `
<div
  className="row justify-content-sm-center"
>
  <div
    className="col"
  >
    <h2>
      Home screen
    </h2>
  </div>
</div>
`;
